src/components/SpiceList.js

```diff
@@ -8,23 +8,36 @@ class SpiceList extends React.Component {
     search: ""
   }
 
+  handleChange = (evt) => {
+    if (evt.type === "checkbox") {
+      this.setState((prevState) => ({
+        fourStarOnly : evt.checked
+      }))
+    } else {
+      this.setState((prevState) => ({
+        search : evt.value
+      }))
+    }
+  }
+
   renderSpices() {
     return this.props.spices
-      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true)
+      .filter(spice => this.state.fourStarOnly ? spice.rating >= 4 : true) //does saying true mean the same as returning the entire object? is it the same saying as saying return everything?
```

Comment:
> does saying true mean the same as returning the entire object? is it the same saying as saying return everything?

When you use `.filter` on an array, it will always return the original object if the callback returns a truthy value. For example:

```js
[1,2,3].filter(num => true) 
// => [1,2,3]

[1,2,3].filter(num => false) 
// => []
```

So in this code, we're doing the same thing essentially - the callback for filter always either returns true or false; and when it returns true, the whole spice object is added to the returned array.

```js
filter(spice => {
  if (this.state.fourStarOnly) {
   return spice.rating >= 4 
  }  else {
    return true
  })
```

Reply:
That's very helpful!! thank you!!!